### ISSUE TITLE
Newline trouble (hello becomes elloh) and pasting trouble

### ISSUE THREAD

**Issue by scootergrisen** [translate/virtaal#3283]
If i translate this string that starts with newline:

    \nto use the same key for text file and swap file

Then i use Alt-Down to copy the string.
Go to the end of the translated string and hit Shift-Home to select the linje.
Then if i write "hello" it becomes "elloh". First letter i type is moves to the end.
Then is i select all with Ctrl+A and paste what i have in the clipboard the "elloh" part is inserted at the end after the text i pasted even though i selected all the text before the paste.

**Comment by rffontenelle** [comment]
Yeah. Even in the latest development version of Virtaal (1.0.0 beta1), newline is sometimes not correctly treated. One thing I notice that always works for me is to remove a newline char from the right side with Backspace - it always remove without issue.

**Comment by dwaynebailey** [comment]
A plausible lead, not confirmed - the autocompletor (same subsystem as #3315) triggers exactly at 4 characters (comp_len=4, so right after typing "hell" of "hello") and defers its suggestion-selection via GLib.idle_add. If your next real keystroke lands while that deferred callback's text selection is still in flight, GTK would replace the selected range with it - a plausible mechanism for character reordering, given the timing-dependent design. Couldn't confirm live this session (an unrelated local environment issue blocked interactive testing) - if you can, try with autocomplete disabled (Preferences > Addons) and see if the newline/reorder issue still happens; that would confirm or rule this out directly.
